admin_organization_test.go

```diff
@@ -0,0 +1,218 @@
+		assert.NoError(t, err)
+
+		// Cannot find deleted org
+		_, err = client.Admin.Organizations.Read(ctx, originalOrg.Name)
+		assert.Error(t, err)
+		assert.EqualError(t, err, ErrResourceNotFound.Error())
+	})
+}
+
+func TestAdminOrganizations_Update(t *testing.T) {
+	skipIfCloud(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	t.Run("it fails to update an organization with an invalid id", func(t *testing.T) {
+		_, err := client.Admin.Organizations.Update(ctx, "", AdminOrganizationUpdateOptions{})
+		require.Error(t, err)
+		assert.EqualError(t, err, ErrInvalidOrg.Error())
+	})
+
+	t.Run("it fails to update an organization with an bad org name", func(t *testing.T) {
+		orgName := fmt.Sprintf("bad-%s", randomString(t))
+		_, err := client.Admin.Organizations.Update(ctx, orgName, AdminOrganizationUpdateOptions{})
+		require.Error(t, err)
+		assert.EqualError(t, err, ErrResourceNotFound.Error())
+	})
+
+	t.Run("fetches and updates organization", func(t *testing.T) {
+		org, orgTestCleanup := createOrganization(t, client)
+		defer orgTestCleanup()
+
+		adminOrg, err := client.Admin.Organizations.Read(ctx, org.Name)
+		assert.NoError(t, err)
+		assert.NotNilf(t, adminOrg, "Org returned as nil")
+
+		accessBetaTools := true
+		isDisabled := false
+		terraformBuildWorkerApplyTimeout := "24h"
+		terraformBuildWorkerPlanTimeout := "24h"
+
+		opts := AdminOrganizationUpdateOptions{
+			AccessBetaTools:                  &accessBetaTools,
+			IsDisabled:                       &isDisabled,
+			TerraformBuildWorkerApplyTimeout: &terraformBuildWorkerApplyTimeout,
+			TerraformBuildWorkerPlanTimeout:  &terraformBuildWorkerPlanTimeout,
+		}
+
+		adminOrg, err = client.Admin.Organizations.Update(ctx, org.Name, opts)
+		assert.NotNilf(t, adminOrg, "Org returned as nil when it shouldn't be.")
+		assert.NoError(t, err)
+
+		assert.Equal(t, accessBetaTools, adminOrg.AccessBetaTools)
+		assert.Equal(t, isDisabled, adminOrg.IsDisabled)
+		assert.Equal(t, terraformBuildWorkerApplyTimeout, adminOrg.TerraformBuildWorkerApplyTimeout)
+		assert.Equal(t, terraformBuildWorkerPlanTimeout, adminOrg.TerraformBuildWorkerPlanTimeout)
+		assert.Equal(t, false, adminOrg.TerraformWorkerSudoEnabled)
+
+		isDisabled = true
+		opts = AdminOrganizationUpdateOptions{
+			IsDisabled: &isDisabled,
+		}
+
+		adminOrg, err = client.Admin.Organizations.Update(ctx, org.Name, opts)
+		assert.NoError(t, err)
+		assert.NotNilf(t, adminOrg, "Org returned as nil when it shouldn't be.")
+
+		assert.Equal(t, adminOrg.IsDisabled, isDisabled)
+
+		isDisabled = false
+		opts = AdminOrganizationUpdateOptions{
+			IsDisabled: &isDisabled,
+		}
+
+		adminOrg, err = client.Admin.Organizations.Update(ctx, org.Name, opts)
+		assert.NoError(t, err)
+		assert.NotNilf(t, adminOrg, "Org returned as nil when it shouldn't be.")
+
+		assert.Equal(t, adminOrg.IsDisabled, isDisabled)
+	})
+}
+
+func adminOrgItemsContainsName(items []*AdminOrganization, name string) bool {
```

Comment:
I can't do `Contains` here because [createOrganization creates an Organization type](https://github.com/hashicorp/go-tfe/blob/master/helper_test.go#L362), and these functions return an `AdminOrganization` type. [iirc, reflection doesn't work ](https://blog.golang.org/laws-of-reflection) across different types that may share similar attributes.

Reply:
Ahhhhhhh derp derp derp yes, got it 👍🏻  Yeah better just to do this than worry about any casting business. Good call!